R/mediate.R

```diff
@@ -0,0 +1,394 @@
+#'   (default: \code{mediator = NULL})
+#'
+#' @param assay.type The name of the assay used for feature-wise mediation analysis.
+#'   (default: \code{assay.type = NULL})
+#' 
+#' @param dim.type The name of the reducedDim used for component-wise mediation analysis.
+#'   (default: \code{dim.type = NULL})
+#'
+#' @param family A specification for the outcome model link function.
+#'    (default: \code{family = gaussian("identity")})
+#' 
+#' @param covariates Name or list of names of colData variables used as covariates
+#'   in the model. (default: \code{covariates = NULL})
+#' 
+#' @param p.adj.method A single character value for selecting adjustment method
+#'   of p-values. Passed to `p.adjust` function. (default: \code{p.adj.method = "BH"})
+#' 
+#' @param add.metadata TRUE or FALSE, should the model metadata be returned.
+#'   (default: \code{add.metadata = FALSE})
+#' 
+#' @param verbose TRUE or FALSE, should execution messages be printed.
+#'   (default: \code{verbose = TRUE})
+#' 
+#' @param ... additional parameters that can be passed to \code{\link[mediation:mediate]{mediate}}.
+#' 
+#' @details
+#' This wrapper of \code{\link[mediation:mediate]{mediate}} for
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#' provides a simple method to analyse the effect of a treatment variable on an
+#' outcome variable found in \code{colData(se)} through the mediation of either
+#' another variable in colData (argument \code{mediator}) or an assay
+#' (argument \code{assay.type}) or a reducedDim (argument \code{dim.type}). Notably,
+#' those three arguments are mutually exclusive.
+#' 
+#' @return 
+#' A \code{data.frame} of p-values, effect sizes and confidence intervals for
+#' the ACMEs and ADEs of every mediator included in the analysis. Columns include:
+#' 
+#' \describe{
+#'   \item{Treatment}{the treatment variable}
+#'   \item{Mediator}{the mediator variable}
+#'   \item{Outcome}{the outcome variable}
+#'   \item{ACME_estimate}{the Average Causal Mediation Effect (ACME) estimate}
+#'   \item{ADE_estimate}{the Average Direct Effect (ADE) estimate}
+#'   \item{ACME_pval}{the p-value for the ACME estimate}
+#'   \item{ADE_pval}{the p-value for the ADE estimate}
+#'   \item{ACME_adjpval}{the adjusted p-value for the ACME estimate}
+#'   \item{ADE_adjpval}{the adjusted p-value for the ADE estimate}
+#'   \item{ACME_CI_lower}{the lower limit of the confidence interval for the ACME estimate}
+#'   \item{ACME_CI_upper}{the upper limit of the confidence interval for the ACME estimate}
+#'   \item{ADE_CI_lower}{the lower limit of the confidence interval for the ADE estimate}
+#'   \item{ADE_CI_upper}{the upper limit of the confidence interval for the ADE estimate}
+#' }
+#'
+#' @name getMediation
+#'
+#' @examples
+#' # Import libraries
+#' library(mia)
+#' library(microbiomeDataSets)
+#' library(scater)
+#' 
+#' # Load dataset
+#' tse <- LahtiWAData()
+#'  
+#' # Convert BMI variable to numeric
```

Comment:
Right. I agree, this is why we have used data sets from `data()` command in all examples. 

In fact, `LahtiWAData()` is available this way as:

```
library(miaTime)
data(hitchip1006)
```

I originally thought that the idea was to use O'Keefe diet swap data (`dietswap()`)? 

Which one works better for a demo?

Reply:
The O'Keefe data, if it is useful, is dietary intervention study. It might be a better example as intervention studies are standard targets for mediation analyses. Anyway, both will do in principle.

Reply:
Now it is imported as suggested by @antagomir

In general, I used hitchip1006 because it comes with pre-calculated alpha diversity, it has more than one confounder so that addition of multiple confounders can be demonstrated (c("sex", "age")), it has more than 2 levels for the treatment variable so the treat.value and control.value functionality can be shown, and unlike okeefe it does not need filtering of only one timepoint

But if okeefe might have some advantage, I can also try that out. Is there a way to load okeefe with `data`?

Reply:
Both are ok for a demo I think.  

I would say that an intervention study, like OKeefe, are more standard for mediation analyses generally speaking but I can see the reasoning.